doc/iterations/2024.3/shapes/improve_resizing_support_for_list_nodes.adoc

```diff
@@ -0,0 +1,38 @@
+= Improves resizing support for list nodes
+
+== Problem
+
+Until now, when a user resizes a list node, it is not functional.
+Only the container rendering is displayed during the resize action, the sub-container being stretched during the next layout.
+And it's not possible to reduce the size of the container (blocked by the size of the sub-elements).
+Sub-container can be resized in any direction, though it is up to the container to guarantee the width.
+
+== Key Result
+
+* When a container is resized, its sub-container must retain the rendering during the entire action.
+* Sub-container must be resizable vertically only.
```

Comment:
Considering only vertically aligned list layout, there are two use cases:

- The child nodes should fill the parent node
- The child nodes should not fill the parent node

Both behavior are required by our end users. Think about it this way, let's consider the following screenshot from the Sirius Desktop gallery:

![](https://eclipse.dev/sirius/images/gallery/modane-1.png)

Consider the nodes with the blue ribbon on the right of the screenshot. You have a node with a label, for example the "StiffenedGas" one containing three child nodes:

- One with the label "StiffenedGas"
- One with the "limit_tension: Real"
- One with the overrides

Those three child nodes are filling their containing node using a list layout. Yet Inside of the second one, we have a list layout with a single element "limit_tension" which is not filling its containing nodes. In your animated gif, you can find this distinction between the parent node and the yellow node for example.

We cannot guess this by ourselves, the specifier must tell us and we must treat both use cases differently.

Reply:
done